### PR TITLE
Add documentation for dIntegrate function.

### PR DESCRIPTION
**Pull request by matthieuvigne**
Fix https://github.com/stack-of-tasks/pinocchio/issues/759

Add some mathematical insight to the documentation of ```integrate```, ```difference``` and ```dIntegrate```.

### CODE REVIEW

File: src/algorithm/joint-configuration.hpp [hunk 1]
```diff
@@ -17,6 +17,9 @@ namespace pinocchio
   /**
    * @brief      Integrate a configuration vector for the specified model for a tangent vector during one unit time
    *
+   * @details This function corresponds to the exponential map of the joint configuration Lie Group.
+   *          Its output can be interpreted as a sum in the Lie algebra \f$ q \oplus v \f$.
```
Comment (jcarpent) [inline src/algorithm/joint-configuration.hpp]: I would say here: `as the sum from the Lie algebra to the joint configuration space`

File: src/algorithm/joint-configuration.hpp [hunk 2]
```diff
@@ -217,6 +229,13 @@ namespace pinocchio
   /**
    * @brief      Computes the Jacobian of a small variation of the configuration vector or the tangent vector into the tangent space at identity.
    *
+   * @details This jacobian has to be interpreted in terms of Lie group, not vector space: as such,
+   *          it is expressed in the tangent space only, not the configuration space.
+   *          Calling \f$ f(q, v) \f$ the integrate function, these jacobians satisfy the following relationships in the
+   *          tangent space:
+   *           - Jacobian relative to q: \f$ f(q \oplus \delta q, v) \ominus f(q, v) = J_q + o(\delta q)\f$.
+   *           - Jacobian relative to v: \f$ f(q, v + \delta v) \ominus f(q, v) = J_v + o(\delta v)\f$.
```
Comment (gabrielebndn) [inline src/algorithm/joint-configuration.hpp]: You are missing the multiplication of the Jacobians by  `\delta q`  and `\delta v` here

File: src/algorithm/joint-configuration.hpp [hunk 3]
```diff
@@ -234,6 +253,13 @@ namespace pinocchio
   /**
    * @brief      Computes the Jacobian of a small variation of the configuration vector or the tangent vector into the tangent space at identity.
    *
+   * @details This jacobian has to be interpreted in terms of Lie group, not vector space: as such,
+   *          it is expressed in the tangent space only, not the configuration space.
+   *          Calling \f$ f(q, v) \f$ the integrate function, these jacobians satisfy the following relationships in the
+   *          tangent space:
+   *           - Jacobian relative to q: \f$ f(q \oplus \delta q, v) \ominus f(q, v) = J_q(q, v) + o(\delta q)\f$.
+   *           - Jacobian relative to v: \f$ f(q, v + \delta v) \ominus f(q, v) = J_v(q, v) + o(\delta v)\f$.
+   *
```
Comment (gabrielebndn) [inline src/algorithm/joint-configuration.hpp]: Same here

**Comment by jmirabel** [comment]
Generally speaking, this doc is a good thing. I think there is a need for a page which summarizes the notations. There are a few tricky points which ought to be highlighted:
1. q0 \oplus (q1 \ominus q0) can be different from q1 (in cases which are however easy to understand).
2. q0 \oplus (v0 + v1) is generally different from (q0 \oplus v0) \oplus v1.
3. You *should* never check equality using q0 == q1 (neither (q0-q1).norm() < thr). This can work if you now what you are doing. If you don't know, better use (q0 \ominus q1).norm() < thr.
4. **You should prefer \oplus and \ominus  operations over + and - ones on configuration vectors**.
5. Moreover, when computing derivatives, you should always think of these operators as functions with their own non-trivial derivatives. It will help you avoid *hard-to-find* wrong derivative (and they are a lot easier to write using the Lie Algebra)

I am sure the list is not exhaustive.

**Comment by jcarpent** [comment]
@matthieuvigne can you address the comments of @gabrielebndn. I will then open a new issue concerning the comments of @jmirabel 
Thanks in advance.

**Comment by matthieuvigne** [comment]
@jmirabel I definitely agree with you on this. It's easy to confuse the joint configuration for a vector space (in particular because in practice, it is implemented as a vector) when you work with only 1DoF joints, and then everything stats to fall apart when you look at more complex kinematics. Viewing everything in terms of Lie group is a lot more powerful, and really well handled by pinocchio, but it takes some practice and it's easy to take the wrong habits.

**Pull request merged**